src/executables/sysrepo-plugind.c

```diff
@@ -444,6 +444,55 @@ apply_plugind_module(int plugin_count, sr_session_ctx_t *sess, struct srpd_plugi
     return rc;
 }
 
+static int
+open_pidfile(const char *pidfile)
+{
+    int pidfd;
+    
+    pidfd = open(pidfile, O_RDWR | O_CREAT, 0640);
+    if (pidfd < 0) {
+        error_print(0, "Unable to open the PID file \"%s\" (%s).", pidfile, strerror(errno));
+        return -1;
+    }
+
+    if (lockf(pidfd, F_TLOCK, 0) < 0) {
+        if ((errno == EACCES) || (errno == EAGAIN)) {
+            error_print(0, "Another instance of the sysrepo-plugind is running.");
+        } else {
+            error_print(0, "Unable to lock the PID file \"%s\" (%s).", pidfile, strerror(errno));
+        }
+        close(pidfd);
+        return -1;
+    }
+
+    return pidfd;
+}
+
+static int
+write_pidfile(int pidfd)
+{
+    char pid[30] = { 0 };
+    int pid_len;
+
+    if (ftruncate(pidfd, 0)) {
+        error_print(0, "Failed to truncate pid file (%s).", strerror(errno));
+        return 0;
```

Comment:
General rule is to return `0` on success and `1` on error so please change ti that way.